utils/marketNextOpen.ts

```diff
@@ -0,0 +1,204 @@
+import {
+	AFTER_HOURS_SYNTHS,
+	COMMODITY_SYNTHS,
+	CurrencyKey,
+	FIAT_SYNTHS,
+	LSE_SYNTHS,
+	TSE_SYNTHS,
+} from 'constants/currency';
+import { addHours, nextMonday, startOfDay, subDays } from 'date-fns';
+import { utcToZonedTime, zonedTimeToUtc } from 'date-fns-tz';
+import moment from 'moment-business-time';
+
+export const forexNextOpen = () => {
+	const currentTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+	const currentTimeInAET = utcToZonedTime(
+		zonedTimeToUtc(new Date(), currentTimezone),
+		'Australia/Sydney'
+	);
+	const forexMarketOpenAET = addHours(startOfDay(nextMonday(subDays(currentTimeInAET, 1))), 9); //9am
+	const forexMarketOpenUTC = zonedTimeToUtc(forexMarketOpenAET, 'Australia/Sydney');
+	const forexMarketOpenLocal = utcToZonedTime(forexMarketOpenUTC, currentTimezone);
+	return forexMarketOpenLocal;
+};
+
+export const usNextOpen = () => {
+	moment.locale('en', {
+		workinghours: {
+			0: null,
+			1: ['09:30:00', '4:00:00'],
+			2: ['09:30:00', '4:00:00'],
+			3: ['09:30:00', '4:00:00'],
+			4: ['09:30:00', '4:00:00'],
+			5: ['09:30:00', '4:00:00'],
+			6: null,
+		},
+		holidays: [
+			// Recurring
+			'*-01-01',
+			'*-07-04',
+			'*-12-25',
+			// 2021
+			'2021-01-18',
+			'2021-02-15',
+			'2021-04-02',
+			'2021-05-31',
+			'2021-07-05',
+			'2021-09-06',
+			'2021-11-25',
+			'2021-12-24',
+			// 2022
+			'2021-01-17',
+			'2021-02-21',
+			'2021-04-15',
+			'2021-05-30',
+			'2021-09-05',
+			'2021-11-24',
+			'2021-12-26',
+			// 2023
+			'2021-01-02',
+			'2021-01-16',
+			'2021-02-20',
+			'2021-04-07',
+			'2021-05-29',
+			'2021-09-04',
+			'2021-11-23',
+		],
+	});
+
+	const currentTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+	const nextOpenET = moment(new Date()).nextWorkingTime().toDate();
+	const nextOpenUTC = zonedTimeToUtc(nextOpenET, 'America/New_York');
+	const nextOpenLocal = utcToZonedTime(nextOpenUTC, currentTimezone);
+	return nextOpenLocal;
+};
+
+export const lseNextOpen = () => {
+	moment.locale('gb', {
+		workinghours: {
+			0: null,
+			1: ['08:00:00', '12:00:00', '12:02:00', '16:30:00'],
+			2: ['08:00:00', '12:00:00', '12:02:00', '16:30:00'],
+			3: ['08:00:00', '12:00:00', '12:02:00', '16:30:00'],
+			4: ['08:00:00', '12:00:00', '12:02:00', '16:30:00'],
+			5: ['08:00:00', '12:00:00', '12:02:00', '16:30:00'],
+			6: null,
+		},
+		holidays: [
+			// Recurring
+			'*-01-01',
+			// 2021
+			'2021-04-02',
```

Comment:
am curious: do the chainlink feeds also track these?

Reply:
i'm not sure, but i know we do pause our synths for these holidays